ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -1025,13 +1027,26 @@ impl TraceBuilder {
             }
             _ => panic!(),
         }
+        let ty = self.handle_type(self.aot_mod.type_(*tyidx))?;
         // Create the constant from the runtime value.
-        let pval = self.promotions[self.promote_idx];
-        self.promote_idx += 1;
-        let c = Const::Int(
-            self.handle_type(self.aot_mod.type_(*tyidx))?,
-            u64::try_from(pval).unwrap(),
-        );
+        let pval = match self.jit_mod.type_(ty) {
+            jit_ir::Ty::Void => unreachable!(),
+            jit_ir::Ty::Integer(width_bits) => {
+                let width_bytes = usize::try_from(*width_bits).unwrap() / 8;
```

Comment:
Do I see it right that the unwrap is safe for systems with a pointer size >=32 bits?

We are unlikely to support anything else, right?

Reply:
I'm not sure these ones are useful to document TBH in the sense that they're kinda obvious. I know we used to say "document all `unwraps`" but this one is really "super weird portability edge cases" where getting things wrong is unlikely; the ones around `from_ne_bytes` are slightly irritating in that they can't actually fail but we're forced to add them anyway.

Reply:
OK. I've been operating in "document all unwraps" mode, but thinking about it, it doesn't really matter. They will crash and we can investigate.

Reply:
I think in the general case it's a useful discipline because it forces us to think about whether we can actually encounter a panic. But I agree that a blanket commenting rule isn't quite right: I think we should document the ones where we have internal-yk invariants which mean we can't encounter a panic. For example a classic one is "we checked X elsewhere so we know it will fit in this type and the unwrap can't fail" -- that's really worth documenting because no future reader will know otherwise! But, say, i32 -> i64 type conversions are really a not-very-interesting artefact of Rust's integer type hierarchy.